pkg/virtual/framework/types.go

```diff
@@ -37,6 +37,13 @@ func (r RootPathResolverFunc) ResolveRootPath(urlPath string, context context.Co
 var _ RootPathResolver = RootPathResolverFunc(nil)
 
 type RootPathResolver interface {
+	// ResolveRootPath returns whether the request should be accepted and served by a given VirtualWorkspace. If accepted,
+	// the prefixToStrip is the prefix that is in-front of the kube-like API surface (including `/clusters/<something>`). E.g. for
+	//
+	//   /services/initializingworkspaces/<initializer>/clusters/<something>/apis/workload.kcp.dev/v1alpha1/workloadclusters
+	//
+	// the prefixToStrip is `/services/initializingworkspaces/<initializer/clusters/<something>`. The returned context holds the APIDomainKey and
+	// the cluster name (e.g. `<something>` here).
```

Comment:
> The returned context holds the APIDomainKey and
	// the cluster name (e.g. `<something>` here).

Just a detail: that's not true of all virtual workspaces, but only of the `DynamicVirtualWorkspace`-based implementations.

Reply:
Can you give a better wording?

Reply:
I'd rather add this sentence in the doc of the `DynamicVirtualWorkspace`
And here we could have something more generic like:

```
The returned context holds whatever information has been extracted from the resolved path, for example the
name of a resource or the logical cluster name.
```

Reply:
That's very unspecific. Isn't there anything required like the APIDomainKey? Who will consume it?

Reply:
The virtual workspace request handler will use the values addedin this context:
- for `workspaces` virtual workspace:
  - the containing workspace
  - the scope
- For the APIExport + Syncer, ... (Based on the DynamicVirtualWorkspace):
  - the APIDomainKey
  - the logical cluster
- For a virtual workspace based on a simple Handler:
  - it can be any information the Resolving action grabs while parsing the accepted virtual workspace root path.